server/services/SsotToRobotParsing/SsotToRobotParser.js

```diff
@@ -0,0 +1,160 @@
+ */
+const isAnRpaInstruction = (currentElement) =>
+  currentElement.rpaTask !== undefined &&
+  currentElement.rpaApplication !== undefined;
+
+/**
+ * @description Checks whether the given element contains rpa parameters
+ * @param {Object} currentElement Element to check
+ * @returns {bool} Value specifies if object contains rpa parameters
+ */
+const rpaParametersExist = (currentElement) =>
+  currentElement.rpaParameters !== undefined &&
+  currentElement.rpaParameters.length !== 0;
+
+/**
+ * @description Checks whether the given element has a successor element
+ * @param {Object} currentElement Element to check
+ * @returns {bool} Value specifies if element has a successor element
+ */
+const successorTasksExist = (currentElement) =>
+  currentElement.successorIds !== undefined &&
+  currentElement.successorIds[0] !== '';
+
+/**
+ * @description Receives an array of all elements and generates the .robot code for the elements recursively.
+ * @param {String} id Id of the element we are looking for
+ * @param {Array} elements All the elements from the SSoT
+ * @param {String} codeToAppend The current code we want to extend
+ * @param {String} previousApplication Name of the rpa application of the previous element
+ * @returns {string} Generated .robot code for the tasks section
+ */
+const writeCodeForElement = (
+  id,
+  elements,
+  codeToAppend,
+  previousApplication
+) => {
+  const currentElement = elements.find((element) => element.id === id);
+  if (isAnRpaInstruction(currentElement)) {
+    if (currentElement.rpaApplication !== previousApplication) {
+      codeToAppend += `${currentElement.name}\n`;
+      previousApplication = currentElement.rpaApplication;
+    }
+    codeToAppend += `  ${currentElement.rpaTask}`;
+    if (rpaParametersExist(currentElement)) {
+      currentElement.rpaParameters.forEach((parameter) => {
+        codeToAppend += `  ${parameter.value}`;
+      });
+    }
+    codeToAppend += '\n';
+  }
+
+  if (successorTasksExist(currentElement)) {
+    // Xor handling is needed here in the future
+    currentElement.successorIds.forEach((successorId) => {
+      codeToAppend = writeCodeForElement(
+        successorId,
+        elements,
+        codeToAppend,
+        previousApplication
+      );
+    });
+
+    return codeToAppend;
+  } else {
+    return codeToAppend;
+  }
+};
+
+/**
+ * @description Receives an array of all elements and generates the .robot code for all RPA Tasks
+ * @param {Array} elements All the elements from the SSoT
+ * @param {Object} metaData MetaData of the robot
+ * @returns {string} Generated .robot code for the tasks section
+ */
+const generateCodeForRpaTasks = (elements) => {
+  const startElement = elements.find(
+    (element) => element.predecessorIds.length === 0
+  );
+  const id = startElement.id;
+  const codeToAppend = '';
+  const previousApplication = 'None';
+
+  const codeForRpaTasks = writeCodeForElement(
+    id,
+    elements,
```

Comment:
why do those parameters need to be defined/declared before? why not just call the function with those values passed directly?

Reply:
I think it's better inline too, since you define it separately beforehand. In this call it is only a passing of parameters.

But hey, let's not waste time over such trivialities here. In the end, someone will run his slightly differently configured Pretty Printer over it and it will break it by itself / make it inline.

@SanJSp @kej-jay